core/src/main/java/com/graphhopper/routing/util/NameSimilarityEdgeFilter.java

```diff
@@ -171,9 +171,17 @@ public final boolean accept(EdgeIteratorState iter) {
         return isJaroWinklerSimilar(pointHint, edgeName);
     }
 
-    private static BBox createBBox(NodeAccess na, EdgeIteratorState edgeState) {
-        return BBox.fromPoints(na.getLat(edgeState.getBaseNode()), na.getLon(edgeState.getBaseNode()),
-                na.getLat(edgeState.getAdjNode()), na.getLon(edgeState.getAdjNode()));
+    private static BBox createBBox(EdgeIteratorState edgeState) {
+        // we should include the entire geometry, see #2319
+        PointList geometry = edgeState.fetchWayGeometry(FetchMode.ALL);
```

Comment:
I've seen this usage pattern a few times now. We create a PointList only to do some logic with each point and throw it away later. Wouldn't it be better to be able to consume them on-the-fly with a visitor pattern?

e.g `edgeState.visitWayGeometry(FetchMode.ALL, visitor);`

Reply:
Yes, maybe. But for this case I wonder if we should simply store the bounding box for each edge.

Reply:
> I wonder if we should simply store the bounding box for each edge.

My europe import logs roughly 92 million edges. That's about 2.8GB of bbox information if we stored minlon,minlat,maxlon,maxlat per edge?

Reply:
Yes, if we used four doubles (4*8=32bytes). But that would be very wasteful. We can definitely reduce precision there (4*4bytes is more than enough), and maybe we can even just store the indices of the point list. Then maybe 4bytes per edge is enough.

Reply:
... and we could probably even somehow restrict this to 'curved' edges only, because for straight edges using the tower node coordinates as bounding box is often sufficient.

Reply:
Implemented visitor access: https://github.com/graphhopper/graphhopper/pull/2321